pkg/apis/openstack/validation/infrastructure.go

```diff
@@ -74,66 +78,104 @@ func ValidateInfrastructureConfigUpdate(oldConfig, newConfig *api.Infrastructure
 }
 
 // ValidateInfrastructureConfigAgainstCloudProfile validates the given InfrastructureConfig against constraints in the given CloudProfile.
-func ValidateInfrastructureConfigAgainstCloudProfile(infra *api.InfrastructureConfig, shootRegion string, cloudProfileConfig *api.CloudProfileConfig, fldPath *field.Path) field.ErrorList {
+func ValidateInfrastructureConfigAgainstCloudProfile(infra *api.InfrastructureConfig, domain, shootRegion string, cloudProfileConfig *api.CloudProfileConfig, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	allErrs = append(allErrs, validateFloatingPoolNameConstraints(cloudProfileConfig.Constraints.FloatingPools, shootRegion, infra.FloatingPoolName, fldPath)...)
+	allErrs = append(allErrs, validateFloatingPoolNameConstraints(cloudProfileConfig.Constraints.FloatingPools, domain, shootRegion, infra.FloatingPoolName, fldPath)...)
 
 	return allErrs
 }
 
-func validateFloatingPoolNameConstraints(fps []api.FloatingPool, region string, name string, fldPath *field.Path) field.ErrorList {
+func validateFloatingPoolNameConstraints(fps []api.FloatingPool, domain, region string, name string, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	_, errs := findFloatingPool(fps, region, name, fldPath.Child("floatingPoolName"))
+	_, errs := FindFloatingPool(fps, domain, region, name, fldPath.Child("floatingPoolName"))
 	allErrs = append(allErrs, errs...)
 	return allErrs
 }
 
-func findFloatingPool(floatingPools []api.FloatingPool, shootRegion, floatingPoolName string, fldPath *field.Path) (*api.FloatingPool, field.ErrorList) {
+// FindFloatingPool finds best match for given domain, region, and name
+func FindFloatingPool(floatingPools []api.FloatingPool, domain, shootRegion, floatingPoolName string, fldPath *field.Path) (*api.FloatingPool, field.ErrorList) {
```

Comment:
@timuthy could you check this as well? You're most familiar with the code, I guess...